autoload/go/highlight_test.vim

```diff
@@ -104,21 +104,45 @@ function! Test_numeric_literal_highlight() abort
   let tests = {
         \ 'lone zero': {'group': 'goDecimalInt', 'value': '0'},
         \ 'integer': {'group': 'goDecimalInt', 'value': '1234567890'},
+        \ 'integerGrouped': {'group': 'goDecimalInt', 'value': '1_234_567_890'},
+        \ 'integerErrorLeadingUnderscore': {'group': 'goDecimalErrorInt', 'value': '_1234_567_890'},
+        \ 'integerErrorTrailingUnderscore': {'group': 'goDecimalErrorInt', 'value': '1_234_567890_'},
+        \ 'integerErrorDoubleUnderscore': {'group': 'goDecimalErrorInt', 'value': '1_234__567_890'},
+        \ 'integerErrorDoubleTrailingUnderscore': {'group': 'goDecimalErrorInt', 'value': '1_234__567_890__'},
         \ 'hexadecimal': {'group': 'goHexadecimalInt', 'value': '0x0123456789abdef'},
+        \ 'hexadecimalGrouped': {'group': 'goHexadecimalInt', 'value': '0x012_345_678_9ab_def'},
         \ 'hexadecimalErrorLeading': {'group': 'goHexadecimalError', 'value': '0xg0123456789abdef'},
         \ 'hexadecimalErrorTrailing': {'group': 'goHexadecimalError', 'value': '0x0123456789abdefg'},
+        \ 'hexadecimalErrorDoubleUnderscore': {'group': 'goHexadecimalError', 'value': '0x__0123456789abdef'},
+        \ 'hexadecimalErrorDoubleTrailingUnderscore': {'group': 'goHexadecimalError', 'value': '0x__0123456789abdef__'},
+        \ 'hexadecimalErrorDoublePrefix': {'group': 'goHexadecimalError', 'value': '0xx0123456789abdef'},
+        \ 'hexadecimalErrorTrailingUnderscore': {'group': 'goHexadecimalError', 'value': '0x0123456789abdef_'},
         \ 'heXadecimal': {'group': 'goHexadecimalInt', 'value': '0X0123456789abdef'},
         \ 'heXadecimalErrorLeading': {'group': 'goHexadecimalError', 'value': '0Xg0123456789abdef'},
         \ 'heXadecimalErrorTrailing': {'group': 'goHexadecimalError', 'value': '0X0123456789abdefg'},
+        \ 'heXadecimalErrorDoublePrefix': {'group': 'goHexadecimalError', 'value': '0XX0123456789abdef'},
         \ 'octal': {'group': 'goOctalInt', 'value': '01234567'},
+        \ 'octalPrefix': {'group': 'goOctalInt', 'value': '0o1234567'},
+        \ 'octalGrouped': {'group': 'goOctalInt', 'value': '0o1_234_567'},
         \ 'octalErrorLeading': {'group': 'goOctalError', 'value': '081234567'},
         \ 'octalErrorTrailing': {'group': 'goOctalError', 'value': '012345678'},
+        \ 'octalErrorDoublePrefix': {'group': 'goOctalError', 'value': '0oo12345678'},
+        \ 'octalErrorDoubleUnderscore': {'group': 'goOctalError', 'value': '0o__12345678'},
+        \ 'octalErrorDoubleTrailingUnderscore': {'group': 'goOctalError', 'value': '0o__12345678__'},
```

Comment:
this test doesn't add value over the `octalErrorDoubleUnderscore`, because this test also has the double underscores in the middle of the number.